### ISSUE TITLE
[KSP2] Resolver.getSymbolsWithAnnotation doesn't seem to work on generated code

### ISSUE THREAD

**Issue by kuanyingchou** [google/ksp#1734]
After generating a class like this:

```
env.codeGenerator.createNewFile(Dependencies(false), "com.example", cls, ext).use {
   it.write("package com.example;".toByteArray())
   it.write("@com.example.A interface $cls {}".toByteArray())
}
```
Calling `Resolver.getSymbolsWithAnnotation()` like this returns an empty sequence in a subsequent round:

```
resolver.getSymbolsWithAnnotation("com.example.A", false)
```

Tested with `2.0.0-Beta3-1.0.18-20240213.110130-21` and 42df0ecc0aeb1431f17bc41759f2d2f90488972e.

**Comment by ting-yuan** [comment]
Can you elaborate on the test case? I tried [this](https://github.com/ting-yuan/ksp/commit/f0b568ce46940165943a3b2f9ffd07b329c4c16c) and it works for me:

**Comment by ting-yuan** [comment]
It also works when the annotation class is generated in the same round of the annotated elements.

**Comment by kuanyingchou** [comment]
Got it. In my record this issue has been fixed after KSP 2.0.0-Beta4-1.0.20-20240308.110129-1 but was present in KSP 2.0.0-Beta4-1.0.19-20240305.110136-6. However I couldn't reproduce this even with KSP 1.0.19 or 1.0.18 today. This could be my setup (Kotlin version?) or even my test code. Sorry for the noise. Let me close it.

**Comment by ting-yuan** [comment]
No worries, and thanks for verifying!
